### ISSUE TITLE
Keep Tabs Sorted by Access Time

### ISSUE THREAD

**Issue by the-a-man-006** [mbnuqw/sidebery#1532]
Can the sorting be made permanent/synchronous?
- So for example, when sorted by access-time (recent to old), clicking on an old tab should refresh the sidebar or drag it up all the way (because it has become more "recent"); likewise, creating new tabs in focus should spawn them at the top (because new tabs are more "recent").

Right now, one has to open the context menu and re--sort-by-access again and again.

Basically, automatically-sort-on-tab-switch BY (creation/access/title/container/url).
A panel-level / per-panel setting maybe.
I, personally, am only interested in auto-sort for last-access though.

Thereupon, you can even do away with:
- new-tabs-open-at (top/bottom) setting.

Would make it the God-Mode for productivity!

**Comment by the-a-man-006** [comment]
https://github.com/mbnuqw/sidebery/issues/643#issuecomment-2002639679

**Comment by mbnuqw** [comment]
I'm narrowing down this feature request to "Keep Tabs Sorted by Access Time" because auto-sorting by url/title is probably a wontfix, auto-sorting by creation time is impossible (Firefox doesn't provide such info) and sorting by container is not implemented yet.

**Comment by vexorian** [comment]
Would it be possible to keep them sorted by *view time* as well? I find that even if there was an auto-sort for access time, it wouldn't work perfectly for me because there's many times I "access" a page without actually downloading the page again. Easiest example would be a long article I scrolled for a couple more paragraphs. But just as frequently,  there are sites that behave as feeds, the page and the access time stay the same but you are browsing more and more 'recent' stuff. Without a refresh, that feed you just refreshed would still have an access time of 7 days ago.

**Comment by the-a-man-006** [comment]
I'm guessing Max would say it's not possible, because it really seems it isn't, because the browser refreshes a page for various "internal" reasons, like when the RAM is too filled up, so it caches it, but then, when you click that tab again, the cache gets so old the browser deems it "expired" so "reloads" the whole page again; plus, the browser probably doesn't provide such an info as "last refresh/reload time" to the extensions, so prolly impossible...

I kinda don't understand the point of sorting tabs by title, URL, container, or even, last-refresh, only auto-sorting by last-access; I'm guessing the people who need those (sort by title/url/container/refresh) need those for very niche reasons, which really don't carry over to daily broader usecases...

Tab previews (implemented recently; https://github.com/mbnuqw/sidebery/issues/301) already allows "reading a paragraph from an old tab" without clicking that tab, thus, without bumping up that tab tree to the top (when auto-sorting by last-access/recent-first). 

For more fine granular control over tab-sorting, like when "reading a paragraph from an old tab by scrolling it down, but without bumping it up to the top", one could resort to manual tab-sorting. To make life easier:

- if there was an action called "bump up" that drags a (sub)tree to the top, above all its siblings, (with permutation, i.e., bumping up the parent, and its parent, and so on),
- then binding that action to the "double-click-on-tab" event would allow you to, when you actually want to "(re)fresh revisit" a tab, not just, "re-read a few more paragraphs without any intention to bump it up", double-click on that tab instead of single-click!

For myself, too, it would suffice if full-blown auto-sort on every tab-switch is too expensive and time-consuming to implement (which I don't think so, cause, sort-by-recent-first is already implemented, and it only requires calling that function whenever the tab-switch event fires or whatever); but it does require a change to the central code, and requires a new setting too, whereas Option 2 only requires adding another assignable action (like the "reload", "duplicate", etc) called "bump-up" to the keybinds/action-binds (like double-click, control-click, etc).

**Comment by otonoton** [comment]
> I kinda don't understand the point of sorting tabs by title, URL, container, or even, last-refresh, only auto-sorting by last-access

I use sort by URL *all* the time, but it gets annoying to have to keep pressing it manually. I'd also really like a "auto-group by URL" and even an "auto-panelize by URL" since I use those too.

> auto-sorting by creation time is impossible

Maybe I'm naive about the internals and how tabs are tracked, but can't you just save the current "now" time when a tab is added to a panel and use that to track the "creation time" for sorting?

**Comment by the-a-man-006** [comment]
This is the only github issue I check every few days to see if it's resolved... Auto-sorting would be a massive game-changer; the only browser, the only extension, with tree-style-tabs, that supports auto-sorting! Would make Firefox (and this extension) a must have for anyone who wishes to be productive and not get lost while switching tabs and surfing, lol.

**Comment by the-a-man-006** [comment]
Please can we have this implemented, please?

Maybe, in 
Sidebery settings, in 
Tab Actions, an entry called:
"On Tab Switch (Left Click)", with a dropdown option called:
"Sort by recent to old".

Thanks.

**Comment by the-a-man-006** [comment]
Please can we have this implemented, please?
